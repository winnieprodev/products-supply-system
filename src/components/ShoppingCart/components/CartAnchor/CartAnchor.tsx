import React from "react";

import * as Styled from "./cartAnchor.styled";

interface OwnProps {
  amount: number;
}
const CartAnchor: React.FC<OwnProps> = ({ amount }) =>
  amount > 0 ? <Styled.Circle>{amount}</Styled.Circle> : null;

export default CartAnchor;
