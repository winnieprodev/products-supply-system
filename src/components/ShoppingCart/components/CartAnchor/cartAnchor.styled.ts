import styled, { keyframes } from "styled-components";
import { primaryColor, textOnPrimaryColor } from "styles/colors";

const popAppear = keyframes`
  from{ 
    transform: scale(0.3);
    opacity: 0;
    width: 0;
  }
  to {
    width: 24px;
    opacity: 1;
    transform: scale(1);
  }
`;

export const Circle = styled.div`
  width: 24px;
  height: 24px;
  border-radius: 50%;
  text-align: center;
  line-height: 24px;
  background: ${primaryColor};
  animation: ${popAppear} ease 0.3s;
  color: ${textOnPrimaryColor};
`;
