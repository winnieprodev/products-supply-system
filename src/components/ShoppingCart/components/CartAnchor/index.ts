export { default as CartAnchor } from "./CartAnchor";
