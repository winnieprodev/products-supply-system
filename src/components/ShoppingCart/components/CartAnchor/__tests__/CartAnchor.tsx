import { render } from "@testing-library/react";

import CartAnchor from "../CartAnchor";

describe("CartAnchor", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for 0 amount", () => {
      const { container } = render(<CartAnchor amount={0} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for 5 amount", () => {
      const { container } = render(<CartAnchor amount={5} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
