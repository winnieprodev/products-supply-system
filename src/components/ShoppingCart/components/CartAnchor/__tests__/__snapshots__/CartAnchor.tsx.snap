// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartAnchor Snapshot tests it matches the snapshot for 0 amount 1`] = `null`;

exports[`CartAnchor Snapshot tests it matches the snapshot for 5 amount 1`] = `
.c0 {
  width: 24px;
  height: 24px;
  border-radius: 50%;
  text-align: center;
  line-height: 24px;
  background: #F25C05;
  -webkit-animation: foLSoj ease 0.3s;
  animation: foLSoj ease 0.3s;
  color: #FFFFFF;
}

<div
  class="c0"
>
  5
</div>
`;
